### PR TITLE
Test OSX on Travis

### PR DESCRIPTION
**Pull request by ca-johnson**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-  - '3.6'
-os:
-  - linux
+jobs:
+  include:
+    - os: linux
+      language: python
+      python: 3.6
+    - os: osx
+      osx_image: xcode12.2 # macOS 10.15.7 Python 3.8.6
```
Comment (ca-johnson) [inline .travis.yml:10]: I originally used pyenv to install 3.6, but this adds an extra 2-3 mins onto the build time that is already 2-3 mins (versus 1 min e2e on linux)

Easy enough to add later, but I think for now we care more about platform coverage than specific minor python versions (since this has never caused an issue historically)

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 before_install:
-  - pip install -U pip
+  - python3 --version
+  - pip3 install -U pip
 install:
-  - pip install -r ./ci/requirements.txt
-  - pip install -r ./python/requirements.txt
+  - python3 -m pip install -r ./ci/requirements.txt
+  - python3 -m pip install -r ./python/requirements.txt
```
Comment (ca-johnson) [inline .travis.yml:16]: Have to be explicit since osx has 2 and 3 installed, but `python` defaults to 2

Throughout the plugin itself we are explicit already

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
+jobs:
+  include:
+    - os: linux
+      language: python
+      python: 3.6
+    - os: osx
+      osx_image: xcode12.2 # macOS 10.15.7 Python 3.8.6
 before_install:
-  - pip install -U pip
+  - python3 --version
```
Comment (ca-johnson) [inline .travis.yml:12]: Driveby to let us see what python version is installed in build logs

File: Makefile [hunk 4]
```diff
@@ -1,6 +1,6 @@
 local_run: vendorize p4d
```
Comment (ca-johnson) [inline Makefile:1]: `rm`, `mkdir` and `cp` on travis osx images are sensitive to having flags at the start.. 🤷  (i.e. `mkdir missing/dir -p` fails while `mkdir -p missing/dir` succeeds)

File: ci/install_osx.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
```
Comment (ca-johnson) [inline ci/install_osx.sh:1]: Pretty much identical functionality to `install_linux.sh`

**Review by petemounce — APPROVED** [review]
LGreatTM!
